core/application/src/state/executor/epoch_change.rs

```diff
@@ -751,14 +754,20 @@ impl<B: Backend> StateExecutor<B> {
         };
 
         // Calculate the epoch end timestamp.
+        let old_epoch_end_timstamp = self.committee_info.get(&epoch).unwrap().epoch_end_timestamp;
         let epoch_time = self.get_epoch_time();
-        let epoch_end_timestamp =
-            self.committee_info.get(&epoch).unwrap().epoch_end_timestamp + epoch_time;
+        let new_epoch_end_timestamp = old_epoch_end_timstamp + epoch_time;
+
+        // Calculate the time interval.
+        let time_interval = (new_epoch_end_timestamp - old_epoch_end_timstamp)
+            .checked_div(self.get_total_intervals())
+            .expect("total_intervals should be validated on initialization");
+        self.time_interval.set(0, time_interval);
```

Comment:
It looks like we only ever use the key `0` for the `time_interval` table. If that's the case, I think we should consider storing the time interval in the `metadata` table.